zebra-chain/src/transaction/serialize.rs

```diff
@@ -356,6 +357,9 @@ impl ZcashSerialize for Transaction {
                 writer.write_u32::<LittleEndian>(5 | (1 << 31))?;
                 writer.write_u32::<LittleEndian>(TX_V5_VERSION_GROUP_ID)?;
 
+                // header: Write the nConsensusBranchId
+                writer.write_u32::<LittleEndian>(u32::from(*consensus_branch_id))?;
```

Comment:
If we switched to NetworkUpgrade, we'd look up the branch id from the network upgrade here.